gen/gen.go

```diff
@@ -128,18 +128,47 @@ func Gen(exercise string, j interface{}, t *template.Template) error {
 		Version: commonMetadata.Version,
 	}, j}
 
+	// Use two potential output files, one for success, and one for format.Source failure.
+	casesFileName := filepath.Join(dirExercise, "cases_test.go")
+	casesErrorFileName := filepath.Join(dirExercise, "cases_test.go-error")
```

Comment:
That is certainly worth considering.  I first sided on making it something that will not be a valid .go file since Format has already failed on it.  But editors can also show syntax errors too quite easily, so making it a .go helps that happen.  With a non-standard extension, I felt free to use the .gitignore with that specific file name quite safely.  If it has a valid .go name, then I didn't feel as comfortable putting _that_ name in a .gitignore file.